Source/scrollrt.cpp

```diff
@@ -740,13 +740,17 @@ void DrawMonsterHelper(const Surface &out, Point tilePosition, Point targetBuffe
 
 	if (leveltype == DTYPE_TOWN) {
 		auto &towner = Towners[mi];
+		assert(towner._tAnimData);
 		int px = targetBufferPosition.x - CalculateWidth2(towner._tAnimWidth);
 		const Point position { px, targetBufferPosition.y };
-		if (mi == pcursmonst) {
+		checkSelection = true;
+		CelClippedDrawTo(out, position, CelSprite(towner._tAnimData, towner._tAnimWidth), towner._tAnimFrame);
+		if (!checkSelection) {

```

Comment:
Why does `false` mean the entity is selected? I feel this is highly unintuitive.

Reply:
I know :D but if I start with true and only change it if there's no selection, there's no guarantee that line memcpy function got called at all ... and atm while the names aren't intuitive, I can be sure it works properly

Reply:
It looks to me like it should still work if you start with false and set it to true instead.